src/builder/specification.md

```diff
@@ -0,0 +1,339 @@
+
+#### Response
+
+- result: `enum`, `"OK" | null`
+- error: code and message set in case an exception happens while registering the validator.
+
+#### Specification
+1. Builder software **MUST** verify `signature` is valid under `pubkey`, otherwise return error `-32005: Invalid Signature`.
+2. Builder software **MUST** respond to requests where `timestamp` is less than or equal to the latest announcement from the validator or more than 1 hour in the future with error `-32007: Invalid timestamp`.
+3. Builder software **MUST** store `feeRecipient` in a map keyed by `pubkey`.
+4. Builder software **MUST** return `result` as `"OK"` if the request succeeds, `null` otherwise.
+
+### `builder_getHeaderV1`
+
+#### Request
+
+- method: `builder_getHeaderV1`
+- params:
+  1. `slot`: `QUANTITY`, 64 Bits - Slot number of the block proposal.
+  2. `pubkey`: `QUANTITY`, 48 Bytes - BLS public key of validator.
+  3. `hash`: `DATA`, 32 Bytes - Hash of execution layer block the proposer will use as the proposal's parent.
+
+#### Response
+
+- result: `object`
+    - `message`: `object`
+        - `header`: [`ExecutionPayloadHeaderV1`](#executionpayloadheaderv1).
+        - `value`: `DATA`, 32 Bytes - Payment in wei that will be paid to the `feeRecipient` account.
+        - `pubkey`: `DATA`, 48 Bytes - BLS public key associated with the builder.
+    - `signature`: `DATA`, 96 Bytes - BLS signature over [`BuilderReceiptV1`](#builderreceiptv1).
+- error: code and message set in case an exception happens while getting the header.
+
+#### Specification
+1. Builder software **SHOULD** respond immediately with the `header` that increases the `feeRecipient`'s balance by the most.
+2. Builder software **MUST** return `-32001: Unknown hash` if the block identified by `hash` is not known.
+3. Builder software **MUST** return `-32002: Unknown validator` if `pubkey` does not map to the validator that is expected to propose at `slot`.
+4. Builder software **MUST** return `-32003: Unknown fee recipient` if the builder does not have a `feeRecipient` mapped to the validator.
+5. Builder software **MAY** set the `feeRecipient` for the block to a different address than the address mapped to the validator so long as a payment equal to `value` is made to `feeRecipient`.
+
+### `builder_getPayloadV1`
+
+#### Request
+
+- method: `builder_getPayloadV1`
+- params:
+  1. `message`: [`BlindBeaconBlock`](#blindbeaconblockv1).
+  2. `signature`: `DATA`, 96 Bytes - BLS signature over [`BlindBeaconBlock`](#blindbeaconblockv1).
+
+#### Response
+
+- result: [`ExecutionPayloadV1`](#executionpayloadv1).
+- error: code and message set in case an exception happens while proposing the payload.
+
+#### Specification
+1. Builder software **MUST** verify that the beacon block's exeuction payload is a matching [`ExecutionPayloadHeaderV1`](#executionpayloadheaderv1) provided from [`builder_getHeaderV1`](#buildergetheaderv1), otherwise the return `-32004: Unknown block`.
```

Comment:
```suggestion
1. Builder software **MUST** verify that the beacon block's execution payload is a matching [`ExecutionPayloadHeaderV1`](#executionpayloadheaderv1) provided from [`builder_getHeaderV1`](#buildergetheaderv1), otherwise the return `-32004: Unknown block`.
```